packages/server/tests/UserSettingsLinkGoogleAccount.test.ts

```diff
@@ -0,0 +1,178 @@
+import { internet, random } from 'faker';
+import test from 'japa';
+import Sinon from 'sinon';
+import sinon from 'sinon';
+import supertest from 'supertest';
+import { BASE_URL, initTestUtils } from './utils/TestUtils';
+
+test.group('User settings link google account group tests', (group) => {
+    const {
+        disconnectEveryRemainingSocketConnection,
+        initSocketConnection,
+        createRequest,
+        createUserAndAuthenticate,
+    } = initTestUtils();
+
+    group.beforeEach(async () => {
+        initSocketConnection();
+        await Database.beginGlobalTransaction();
+    });
+
+    group.afterEach(async () => {
+        await disconnectEveryRemainingSocketConnection();
+        sinon.restore();
+        await Database.rollbackGlobalTransaction();
+    });
+
+    test('It should fail to link a google account as user is not authenticated', async () => {
+        await supertest(BASE_URL)
+            .post('/me/link-google-account')
+            .send({
+                userGoogleAccessToken: random.word(),
+            } as LinkGoogleAccountRequestBody)
+            .expect(401);
+    });
+
+    test('It should fail to link a google account as user doesnot have verified her email', async () => {
+        const request = createRequest();
+
+        const emailIsNotConfirmed = true;
+        await createUserAndAuthenticate(request, emailIsNotConfirmed);
+
+        await request
+            .post('/me/link-google-account')
+            .send({
+                userGoogleAccessToken: random.word(),
+            } as LinkGoogleAccountRequestBody)
+            .expect(403);
+    });
+
+    test('It should fail to link a google account as user already has a linked google account', async () => {
+        const request = createRequest();
+
+        const emailIsNotConfirmed = true;
+        const user = await createUserAndAuthenticate(
+            request,
+            emailIsNotConfirmed,
+        );
+        user.googleID = random.word();
+        await user.save();
+
+        await request
+            .post('/me/link-google-account')
+            .send({
+                userGoogleAccessToken: random.word(),
+            } as LinkGoogleAccountRequestBody)
+            .expect(403);
+    });
+
+    test('It should fail to link a google account as body is invalid', async () => {
+        const request = createRequest();
+
+        await createUserAndAuthenticate(request);
+
+        await request.post('/me/link-google-account').expect(500);
+    });
+
+    test('It should fail to link a google account as body is missing', async () => {
+        const request = createRequest();
+
+        await createUserAndAuthenticate(request);
+
+        await request
+            .post('/me/link-google-account')
+            .send({
+                userGoogleAccessToken: random.word(),
+            } as LinkGoogleAccountRequestBody)
+            .expect(500);
+    });
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏I don't understand the purpose of this test. What is it for?